pkg/provider/onepassword/onepassword.go

```diff
@@ -340,14 +355,21 @@ func (provider *ProviderOnePassword) GetSecret(_ context.Context, ref esv1beta1.
 		return nil, err
 	}
 
-	// handle files
+	// handle Document secret type
 	if item.Category == documentCategory {
-		// default to the first file when ref.Property is empty
-		return provider.getFile(item, ref.Property)
+		propertyType, property := getObjType(ref.Property, filePrefix)
+		if propertyType == fieldPrefix {
+			return provider.getField(item, property)
+		}
+		return provider.getFile(item, property)
 	}
 
-	// handle fields
-	return provider.getField(item, ref.Property)
+	// handle other secret types
+	propertyType, property := getObjType(ref.Property, fieldPrefix)
+	if propertyType == filePrefix {
+		return provider.getFile(item, property)
+	}
+	return provider.getField(item, property)
```

Comment:
Change the getObjType to this:

```go
// Clean property string by removing property prefix if needed.
func getObjType(documentType onepassword.ItemCategory, property string) (string, string) {
	if strings.HasPrefix(property, fieldPrefix+prefixSplitter) {
		return fieldPrefix, property[6:]
	}
	if strings.HasPrefix(property, filePrefix+prefixSplitter) {
		return filePrefix, property[5:]
	}

	if documentType == documentCategory {
		return filePrefix, property
	}

	return fieldPrefix, property
}
```


then you can call it like this:
```go
	// handle other secret types
	propertyType, property := getObjType(item.Category, ref.Property)
	if propertyType == filePrefix {
		return provider.getFile(item, property)
	}
	return provider.getField(item, property)
```

end remove the other one that checks for the item.Category.

Reply:
Fixed